game-headed/src/test/java/org/triplea/game/client/ui/javafx/screens/MapSelectionTest.java

```diff
@@ -0,0 +1,111 @@
+package org.triplea.game.client.ui.javafx.screens;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.clearInvocations;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.engine.framework.ui.GameChooserEntry;
+import java.lang.reflect.Field;
+import java.util.List;
+import javafx.scene.Node;
+import javafx.scene.layout.Pane;
+import javafx.scene.web.WebEngine;
+import javafx.scene.web.WebView;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.triplea.game.client.parser.GameDetector;
+import org.triplea.game.client.ui.javafx.screen.ScreenController;
+import org.triplea.game.client.ui.javafx.util.FxmlManager;
+
+public class MapSelectionTest {
+
+  private GameDetector gameDetector;
+  private MapSelection mapSelection;
+
+  @BeforeEach
+  void setUp() {
+    gameDetector = mock(GameDetector.class);
+    mapSelection = new MapSelection(gameDetector, String::concat);
+  }
+
+  @Test
+  void testOnShowTriggersLoadingCorrectly() {
+    mapSelection.onShow();
+
+    verify(gameDetector).discoverGames(any());
+
+    clearInvocations(gameDetector);
+
+    mapSelection.onShow();
+
+    verify(gameDetector, never()).discoverGames(any());
+  }
+
+  @Test
+  void testReturnsCorrectNode() {
+    final Node node = mock(Node.class);
+    mapSelection.setRoot(node);
+
+    assertThat(mapSelection.getNode(), is(equalTo(node)));
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  void testBackKeyWorks() {
+    final ScreenController<FxmlManager> manager = mock(ScreenController.class);
+
+    mapSelection.connect(manager);
+
+    mapSelection.backToGameSelection();
+
+    verify(manager).switchScreen(FxmlManager.GAME_SELECTION_CONTROLS);
+  }
+
+  @Test
+  void testCloseButtonClosesDetails() {
+    final Node node = mock(Pane.class);
+    mapSelection.setPreviewContainer(node);
+
+    mapSelection.closeDetails();
+
+    verify(node).setVisible(false);
+  }
+
+  @Test
+  void testDetailsButtonShowsDetails() throws Exception {
```

Comment:
side-note and nit: IMO the `test` prefix can be removed from the test method names without losing anything. `detailsButtonShowsDetails()` is just as descriptive IMO.

Of more importance though, this is a somewhat lengthy test. It might benefit from extracting the setup code into helper methods. To explain through a rough example:
```
givenPreviewWithWebEngine(webEngine);
givenMapSelectionWithContainerAndWindow(previewContainer, previewWindow);
givenGameDataWithMapName("Test name");

:
:

mapSelection.showDetails();

    verify(previewContainer).setVisible(true);
    verify(webEngine).loadContent("Test Notes\n Test Name \n");
```

Though, IMO it would be even better if we could have decompose the logic a bit more. Notably so that 'showDetails' returns a value (or uses another object that returns a value for us). This way we can test the results of a return value and not need to use `verify`, which is essentially testing for side-effects.  I think perhaps it indicates the underlying logic would benefit from strategy objects to perform the logic rather than owning the logic itself. 

```